src/qililab/platform/platform.py

```diff
@@ -567,6 +591,41 @@ def __str__(self) -> str:
         """
         return str(YAML().dump(self.to_dict(), io.BytesIO()))
 
+    def execute_anneal_program(
+        self, anneal_program_dict: list[dict[str, dict[str, float]]], transpiler: Callable, averages=1
```

Comment:
Refer to https://github.com/qilimanjaro-tech/qililab/pull/767#discussion_r1713447189. Ofc this might be addressed in a separate pr building on top of this.

> In QProgram, the parameter for the number of averages is called shots, following other frameworks that use the same name. Should we rename averages to shots this to be consistent?

Using shots when we are actually running averages is misleading. Other frameworks allow you to recover statistics about shots (perhaps not all of them) which cannot be accessed when running averages (eg standard deviation). I think it's much more clear to name it average since we are actually running an average, and not retrieving each individual "shot"

Reply:
Oh maybe I messed up the answers, but this was answered here I believe https://github.com/qilimanjaro-tech/qililab/pull/767#discussion_r1713813800 (tldr added a task for this)
Also related to this https://github.com/qilimanjaro-tech/qililab/pull/767#discussion_r1716738156 (tldr yes we should do that, I wouldnt do it this sprint though)